packages/animations/lib/src/open_container.dart

```diff
@@ -0,0 +1,636 @@
+  ///
+  /// When it is included, it may be visible or not according to [isVisible].
+  bool get isInTree => _placeholderSize == null;
+
+  @override
+  Widget build(BuildContext context) {
+    if (_placeholderSize != null) {
+      return SizedBox.fromSize(size: _placeholderSize);
+    }
+    return Opacity(
+      opacity: _visible ? 1.0 : 0.0,
+      child: widget.child,
+    );
+  }
+}
+
+class _OpenContainerRoute extends ModalRoute<void> {
+  _OpenContainerRoute({
+    @required Color closedColor,
+    @required this.openColor,
+    @required double closedElevation,
+    @required this.openElevation,
+    @required ShapeBorder closedShape,
+    @required this.openBuilder,
+    @required this.closedBuilder,
+    @required this.hideableKey,
+    @required this.closedBuilderKey,
+  })  : assert(closedColor != null),
+        assert(openColor != null),
+        assert(closedElevation != null),
+        assert(openElevation != null),
+        assert(closedShape != null),
+        assert(openBuilder != null),
+        assert(closedBuilder != null),
+        assert(hideableKey != null),
+        assert(closedBuilderKey != null),
+        _elevationTween = Tween<double>(
+          begin: closedElevation,
+          end: openElevation,
+        ),
+        _shapeTween = ShapeBorderTween(
+          begin: closedShape,
+          end: const RoundedRectangleBorder(),
+        ),
+        _colorTween = _FlippableTweenSequence<Color>(<TweenSequenceItem<Color>>[
+          TweenSequenceItem<Color>(
+            tween: ColorTween(begin: closedColor, end: Colors.white),
+            weight: 4 / 12,
+          ),
+          TweenSequenceItem<Color>(
+            tween: ColorTween(begin: Colors.white, end: openColor),
+            weight: 8 / 12,
+          ),
+        ]);
+
+  final Color openColor;
+  final double openElevation;
+  final OpenContainerBuilder closedBuilder;
+  final OpenContainerBuilder openBuilder;
+  final GlobalKey<_HideableState> hideableKey;
+  final GlobalKey closedBuilderKey;
+
+  final Tween<double> _elevationTween;
+  final ShapeBorderTween _shapeTween;
+  final _FlippableTweenSequence<Color> _colorTween;
+
+  final _FlippableTweenSequence<double> _closedOpacityTween =
+      _FlippableTweenSequence<double>(<TweenSequenceItem<double>>[
+    TweenSequenceItem<double>(
+      tween: Tween<double>(begin: 1.0, end: 0.0),
+      weight: 4 / 12,
+    ),
+    TweenSequenceItem<double>(
+      tween: ConstantTween<double>(0.0),
+      weight: 8 / 12,
+    ),
+  ]);
+  final _FlippableTweenSequence<double> _openOpacityTween =
+      _FlippableTweenSequence<double>(<TweenSequenceItem<double>>[
+    TweenSequenceItem<double>(
+      tween: ConstantTween<double>(0.0),
+      weight: 4 / 12,
+    ),
+    TweenSequenceItem<double>(
+      tween: Tween<double>(begin: 0.0, end: 1.0),
+      weight: 8 / 12,
+    ),
+  ]);
+
+  final GlobalKey _openBuilderKey = GlobalKey();
+
+  // Defines the position of the container on screen.
```

Comment:
container => OpenContainer (?)

And it's not really the "screen", it's the bounds of the enclosing navigator (which is often the same size as the screen of course).